### PR TITLE
Add tests to grid

### PR DESCRIPTION
**Pull request by crewtool**
# Description
This PR adds basic tests to grid. We need them to make sure grid refactor does not affect us. 
To be more confident that grid works fine after refactor I added tests checking if arguments for remote cluster training are the same,
Grid functionality did not change.

These tests are temporary. We need to make sure we do not change grid behaviour.

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: lizrd/scripts/grid.py [hunk 1]
```diff
@@ -29,27 +29,25 @@
...
     else:
         raise ValueError("config path point to a .yaml")
 
     for config in configs:
-        config["params"]["git_branch"] = args.git_branch
-        config["params"]["path_to_entry_config"] = args.config_path
-        config["params"]["all_config_paths"] = ",".join(all_config_paths)
+        config["params"]["git_branch"] = git_branch
+        config["params"]["path_to_entry_config"] = config_path
+        config["params"]["all_config_paths"] = ",".join(sorted(list(all_config_paths)))
```
Comment (crewtool) [inline lizrd/scripts/grid.py:50]: all_config_paths have nondeterministic order

Comment (Krystian-Krol) [reply lizrd/scripts/grid.py:50]: ja bym te logikę sortowania wywalił do funkcji `load_with_inheritance` bo to tam robimy seta. Taka opcja sprawia że mamy rozbitą logikę tworzenia `all_config_paths` na dwie różne funkcje.

File: lizrd/scripts/grid.py [hunk 2]
```diff
@@ -86,7 +84,7 @@
 
     check_for_argparse_correctness(grid)
 
-    if not CLUSTER_NAME == MachineBackend.LOCAL:
+    if (not CLUSTER_NAME == MachineBackend.LOCAL) and (not skip_confirmation):
```
Comment (Krystian-Krol) [inline lizrd/scripts/grid.py:87]: błagam zmień `not ==` na `!=`

File: lizrd/scripts/grid.py [hunk 3]
```diff
@@ -199,18 +196,52 @@
...
+    )
+    if CLUSTER_NAME != MachineBackend.LOCAL:
+        for i, experiment in enumerate(experiments):
+            subprocess_args, job_name = experiment
+            print(f"running experiment {i} from {job_name}...")
+            PROCESS_CALL_FUNCTION(subprocess_args)
+            sleep(5)
+            if interactive_debug_session:
+                print(
+                    "Ran only the first experiment in (interactive mode or local run). Aborting..."
```
Comment (Krystian-Krol) [inline lizrd/scripts/grid.py:238]: W szczególności tutaj ten print już jest nieaktualny, bo jedynym powodem może być interactive mode

**Review by Krystian-Krol — APPROVED** [review]
Seems legit + small nits

File: lizrd/test/test_lr_grid.yaml
Comment (Krystian-Krol) [inline lizrd/test/test_lr_grid.yaml:1]: Also, I would move this file to `config/test`

File: lizrd/scripts/grid.py (hunk 2 above)
Comment (crewtool) [reply lizrd/scripts/grid.py:87]: ah, jasne

File: lizrd/test/test_lr_grid.yaml
Comment (crewtool) [reply lizrd/test/test_lr_grid.yaml:1]: config/test are run in in integrated test, which is not the purpose here

**Pull request merged**